testsuite/features/step_definitions/common_steps.rb

```diff
@@ -1413,10 +1413,69 @@ def server_secret
 And(/^I check the Cobbler parameter "([^"]*)" with value "([^"]*)" in the isolinux.cfg$/) do |param, value|
   tmp_dir = "/var/cache/cobbler/buildiso"
   result, code = $server.run("cat #{tmp_dir}/isolinux/isolinux.cfg | grep -o #{param}=#{value}")
-  raise "error during veryfying isolinux.cfg parameter for Cobbler buildiso.\nLogs:\n#{result}" if code.nonzero?
+  raise "error during verifying isolinux.cfg parameter for Cobbler buildiso.\nLogs:\n#{result}" if code.nonzero?
 end
 
 When(/^I cleanup after Cobbler buildiso$/) do
   result, code = $server.run("rm -Rf /var/cache/cobbler")
   raise "error during Cobbler buildiso cleanup.\nLogs:\n#{result}" if code.nonzero?
 end
+
+When(/^I reboot server through SSH$/) do
+  node = get_target('server')
+  temp_server = twopence_init("ssh:#{$server.public_ip}")
+  temp_server.extend(LavandaBasic)
+  temp_server.run("reboot > /dev/null 2> /dev/null &")
+  reboot_timeout = 300
+
+  check_shutdown($server.public_ip, reboot_timeout)
+  check_restart($server.public_ip, temp_server, reboot_timeout)
+
+  repeat_until_timeout(timeout: reboot_timeout, message: "Spacewalk didn't come up") do
+    out, code = temp_server.run('spacewalk-service status', check_errors: false, timeout: 10)
+    if !out.to_s.include? "dead" and out.to_s.include? "running"
+      log "Server spacewalk service is up"
+      break
+    end
+    sleep 1
+  end
+end
+
+Then(/^I change server short hostname from hosts and hostname files$/) do
+  old_hostname = $server.hostname
+  new_hostname = old_hostname + '2'
+  log "New short hostname: #{new_hostname}"
+
+  $server.run("sed -i 's/#{old_hostname}/#{new_hostname}/g' /etc/hostname &&
+  echo '#{$server.public_ip} #{$server.full_hostname} #{old_hostname}' >> /etc/hosts &&
```

Comment:
beware an entry might already exist in /etc/hosts for the same IP :-( 

I would proceed slightly differently: add a second ip address to the network card

and then have name resolution (in /etc/hosts if you insist, in dns otherwise) for that other ip address

that would avoid those potential conflicts